pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,384 @@
+		// Short: f.Localizer.MustLocalize("registerCluster.cmd.shortDescription"),
+		Short:   "registerCluster.cmd.shortDescription",
+		Long:    "registerCluster.cmd.longDescription",
+		Example: "registerCluster.cmd.example",
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	// TODO add Localizer and flags
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	// this flag will allow the user to pass the cluster id as a flag
+	opts.selectedClusterId = "123"
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", "cluster id")
+	//flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("registerCluster.flag.clusterId"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) error {
+	getListClusters(opts)
+	runClusterSelectionInteractivePrompt(opts)
+	getMachinePoolList(opts)
+	selectAccessPrivateNetworkInteractivePrompt(opts)
+	registerClusterWithKasFleetManager(opts)
+
+	return nil
+}
+
+func getListClusters(opts *options) error {
+	// ocm client connection
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, cc, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+
+	resource := client.Clusters().List()
+	response, err := resource.Send()
+	if err != nil {
+		return err
+	}
+	clusters := response.Items()
+	var cls = []v1.Cluster{}
+	for _, cluster := range clusters.Slice() {
+		// TODO the cluster must be multiAZ
+		//if cluster.State() == "ready" && cluster.MultiAZ() == true {
+		if cluster.State() == "ready" {
+			cls = append(cls, *cluster)
+		}
+	}
+	opts.clusterList = cls
+	defer cc()
+	//defer conn.Close()
+	return nil
+}
+
+func runClusterSelectionInteractivePrompt(opts *options) error {
+	// get rid of this
+	clusterListString := make([]string, 0)
+	for _, cluster := range opts.clusterList {
+		clusterListString = append(clusterListString, cluster.Name())
+	}
+
+	// TODO add page size and Localizer
+	prompt := &survey.Select{
+		Message: "Select the ready cluster to register",
+		Options: clusterListString,
+	}
+
+	var selectedClusterName string
+	err := survey.AskOne(prompt, &selectedClusterName)
+	if err != nil {
+		return err
+	}
+
+	// get the desired cluster
+	for _, cluster := range opts.clusterList {
+		if cluster.Name() == selectedClusterName {
+			opts.selectedCluster = cluster
+		}
+	}
+	return nil
+}
+
+func parseDNSURL(opts *options) (string, error) {
+	clusterIngressDNSName := opts.selectedCluster.Console().URL()
+	if len(clusterIngressDNSName) == 0 {
+		return "", fmt.Errorf("url is empty")
+	}
+	return strings.SplitAfter(clusterIngressDNSName, ".apps.")[1], nil
+}
+
+// TODO this function should be split the ocm call and the response flow logic
+func getMachinePoolList(opts *options) error {
+	// ocm client connection
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, cc, err := conn.API().OCMClustermgmt()
```

Comment:
Unless otherwise I am missing something, this `connection` assumes that KFM & OCM are deployed on the same gateway and are both using RH sso prod.
This is not always true as indicated by https://github.com/redhat-developer/app-services-cli/pull/1795#discussion_r1084149057 - we should create a dedicated OCM connection  as done in `CreateOCMConnection()` - later on the way this connection is created can be API driven configurable